plugins/derive/g_derive.mlg

```diff
@@ -8,13 +8,21 @@
 (*         *     (see LICENSE file for the text of the license)         *)
 (************************************************************************)
 
+{
+
 open Stdarg
 
+}
+
 DECLARE PLUGIN "derive_plugin"
 
+{
+
 let classify_derive_command _ = Vernacexpr.(VtStartProof ("Classic",Doesn'tGuaranteeOpacity,[]),VtLater)
 
-VERNAC COMMAND EXTEND Derive CLASSIFIED BY classify_derive_command
+}
+
+VERNAC COMMAND EXTEND Derive CLASSIFIED BY { classify_derive_command }
 | [ "Derive" ident(f) "SuchThat" constr(suchthat) "As" ident(lemma) ] ->
-     [ Derive.start_deriving f suchthat lemma ]
+  { Derive.start_deriving f suchthat lemma }
```

Comment:
This allows to parse OCaml code without having to write an OCaml parser, using the same kind of trick as in e.g. melt. We have to wrap quotes inside delimiters that are well-parenthesized in the object language, so that it is just a matter of keeping a count of how many delimiters are opened so far (and handling a few constructs like strings and comments).

Reply:
I guess the question was about why moving from square brackets to curly brackets to delimit OCaml code. I guess that the reason is by consistency with the newly introduced curly brackets in `mlg` files. OK, you could ask why not to use square brackets in `mlg` files by consistency with `VERNAC EXTEND`. Then, I would guess that this is by consistency with `ocamlyacc`. The hysterical paths followed by syntax are often tortuous...

Reply:
You have to change at least one of the uses of brackets into something else, because you cannot use it both for quotes and as a token. I went for braces for OCaml code because this was minimal in changes regarding camlp5-style grammar extension.

Reply:
It was about both. And the answers are perfectly clear, thanks :slightly_smiling_face: